### DISCUSSION TITLE
configuration.yaml latest beta

### DISCUSSION THREAD

**Discussion by Tinus78** [djansen1987/SAJeSolar#48] [Q&A]
I added the latest beta version (B6) to my HASS (my solar installation has multiple plants), but the yaml code in the readme didn't work. (no valid configuration error)
Does the latest beta require a different yaml code?

Edit: removed todayalarm & systempower the errors are gone

`
  - platform: saj_esolar
    username: *****
    password: *****
    plant_id: 0 # Default is 0. Typically ordered in the same way as they are listed in the eSolar app
    resources:
      - nowPower
      - runningState
      - todayElectricity
      - monthElectricity
      - yearElectricity
      - totalElectricity # Energy -> Solar production
      - todayGridIncome
      - income
      - lastUploadTime
      - totalPlantTreeNum
      - totalReduceCo2
      #- todayAlarmNum
      - isAlarm # Yes / No
      - status
      - plantuid
      - currency
      - address
      - isOnline
      - peakPower
      #- systemPower # Installed capacity
`

**Comment by djansen1987** [comment]
Hi Please use the latest stable, the beta has indeed an different yaml config but also has some issues.

**Reply by djansen1987** [reply]
if you would like to test the latest beta, lose the complete resource part in your yaml config. Trying out autoconfiguration based on your model

**Reply by Tinus78** [reply]
I am trying to read out my 6 plants, does the latest stable release support multiple plants?

**Reply by djansen1987** [reply]
it should, but did not test my self. One of the latest features added by a contributer

**Reply by djansen1987** [reply]
they are also workig on a new addon which might fit your needs more, please have a read here: https://github.com/djansen1987/SAJeSolar/discussions/45

**Reply by Tinus78** [reply]
There was a little bug in the code, all the sensor values were written in the same entities. I changed the code:
from:
`
self._attr_unique_id = f"{SENSOR_PREFIX}_{self._type}" 

to:
`
self._attr_unique_id = f"{SENSOR_PREFIX}_{self._type}_{self.plant_id}" #_{self.plant_id} added
`
now every plant_id gets unique entities

**Reply by Tinus78** [reply]
and changed
from:
'
self._attr_name = SENSOR_PREFIX + self.entity_description.name
'
to
'
self._attr_name = SENSOR_PREFIX + self.entity_description.name +str(self.plant_id)
'

Now every entity has a unique name and key:

esolar gridPower1
sensor.esolar_gridpower1

**Reply by djansen1987** [reply]
is your question answered? If so could you please mark it as Answered.
